lib/smart_answer_flows/benefit-cap-calculator.rb

```diff
@@ -92,21 +99,52 @@ def define
           if response == "none"
             outcome :outcome_not_affected
           else
-            question BenefitCapCalculatorFlow.next_benefit_amount_question(config.questions, benefit_types)
+            question BenefitCapCalculatorFlow.next_benefit_amount_question(config.questions(chosen_cap), benefit_types)
+          end
+        end
+      end
+
+      #Q4 future flow
+      checkbox_question :receiving_non_exemption_benefits_future? do
+        config.benefits(:future).keys.each do |benefit|
+          option benefit
+        end
+
+        next_node_calculation :benefit_types do |response|
+          response.split(",").map(&:to_sym)
+        end
+
+        precalculate :benefit_options do
+          config.descriptions(chosen_cap).merge(none_above: "None of the above")
+        end
+
+        calculate :total_benefits do
+          0
+        end
+
+        calculate :benefit_cap do
+          0
+        end
```

Comment:
@leenagupte:

The [Storing data for later use][1] documentation & example flow explain the difference between the various types of calculation block. Essentially, a value stored in a `precalculate` block is available in other blocks within the same question and within the template for that question; whereas a value stored in a `calculate` block is only available in blocks in subsequent questions and their templates.

Since in this case you would be moving the `benefit_options` block into a previous question, there would no longer be any need for it to be a `precalculate` block - it can just be a `calculate` block. And I think the latter would be less confusing.

I know this flow DSL stuff is really confusing - it took @chrisroos & I ages to get our heads round it - hence that documentation page! One of our longer-term goals is to simplify/remove this DSL, but for now we're stuck with it.

[1]: https://github.com/alphagov/smart-answers/blob/master/doc/storing-data.md